### DISCUSSION TITLE
Flask server in production

### DISCUSSION THREAD

**Discussion by kkavitha** [sse-secure-systems/connaisseur#177] [Q&A] [answered]
Hi!

I have a question: When deploying connaisseur through helm onto a kubernetes cluster, is the deployed server a flask server or is it behind some other service? I saw that the docker image runs `python -m connaisseur`. I noticed that the `__main__.py` uses `APP.run`. Is this run synonymous to a `flask run` command? When looking at documentation found here, https://flask.palletsprojects.com/en/2.0.x/tutorial/deploy/#run-with-a-production-server, there is mention that `flask run` is not production ready.
If this is not using the flask server, which server is it using? Does deploying this in a cluster negate this point?

I am pretty new to this and would appreciate any clarification! Thanks!

**Comment by xopham** [comment, marked as answer]
Hi @kkavitha :wave: you're right. Connaisseur is using a Flask server. As discussed in #11, this is mostly not recommended for performance reasons which we consider okay, as it should not be externally exposed and typically not experience high load. For now, Flask offers a decent tradeoff between performance and resource usage.
However, we are currently testing other servers via load testing and resource constraints to identify a more performant solution.

**Reply by kkavitha** [reply]
Thank you for the response!
